src/systemd/sd-bus-vtable.h

```diff
@@ -57,17 +61,20 @@ struct sd_bus_vtable {
         union {
                 struct {
                         size_t element_size;
+                        sd_bus_vtable_features features;
```

Comment:
enums a slightly problematic when exposed in ABIs, since when you add a new field they might increase in size, and that's not pretty, as it will propagate size changes through all its uses in a non-obvious way. My proposal would be to maybe just use a bitfield in this case, i.e. just use `unsigned parameter_names:1;` as field here. Of course, this means that the size of the struct might change too if we add many more flags, but at least its obvious now, as the space use is directly visible in the structure, and the types don't magically change size.

Reply:
(and sorry for suggesting something else first, I forgot the ABI implications for an enum that might grow there...)

(other alternative here is to just use uint64_t or so as a flags field, i.e. a fixed type field, i'd be fine with tha too)

Reply:
I would prefer to just change the type here to uint64_t instead of the enum, but leave the enum to list the flags. Is that ok?

Reply:
yes, but please make the enum anonymous

Reply:
done, and also renamed the flag as _SD_BUS_VTABLE_PARAM_NAMES (with heading underscore). I believe that also in this project it means: internal, but exposed because used somewhere in exported macros.